File: Flutter/hksy_flutter/lib/pages/coin/api/coinapi.dart
import 'package:hksy_flutter/public/networking.dart';

class CoinApi {
  static giftCoinTransfer(userId, balance, finish) {
    Networking.requestBase("giftCoinTransfer", finish, params: {
      "userId": userId,
      "balance": balance,
    });
  }

  static vipgiftCoinTransfer(userId, balance, finish) {
    Networking.requestBase("vipgiftCoinTransfer", finish, params: {
      "userId": userId,
      "balance": balance,
    });
  }

  static queryBalance(userId, page, limit, finish) {
    Networking.requestBase("queryBalance", finish, params: {
      "userId": userId,
      "page": page,
      "limit": limit,
    });
  }

  static transfer(userId, phone, paymentCode, balance, finish) {
    Networking.requestBase("transfer", finish, params: {
      "ownerUserId": userId,
      "phone": phone,
      "paymentCode": paymentCode,
      "balance": balance,
    });
  }

  static offlineRecharge(userId, voucherPath, balance, finish) {
    Networking.requestBase("offlineRecharge", finish, params: {
      "userId": userId,
      "voucherPath": voucherPath,
      "balance": balance,
    });
  }

  static submitUSDTWithdraw(
      userId, coinnumber, arrivalamount, usdtfee, outaddress, finish) {
    Networking.requestBase("offlineRecharge", finish, params: {
      "userId": userId,
      "coin_number": coinnumber,
      "arrival_amount": arrivalamount,
      "usdt_fee": usdtfee,
      "out_address": outaddress,
    });
  }

  static withdrawCash(userId, cashNumber, paymentCode, finish) {
    Networking.requestBase("withdrawCash", finish, params: {
      "userId": userId,
      "cashNumber": cashNumber,
      "paymentCode": paymentCode,
    });
  }
}

File: UNI-APP/hksy_uni_3.0/pages/coin/api/coinapi.js
import networking from "../../../public/networking.js"

// 获取当前用户的账户明细(分页)
function queryBalance(userId, page, limit, completion) {
	networking.functionRequest("/queryBalance", null, {
		"userId": userId,
		"current": page,
		"size": limit
	}, null, completion)
}

// 金币划转
function transfer(params, completion) {
	networking.functionRequest("/transfer", null, params, null, completion)
}

// 线下充值申请
function offlineRecharge(userId, voucherPath, balance, completion) {
	networking.functionRequest("/offlineRecharge", null, {
		"userId": userId,
		"voucherPath": voucherPath,
		"balance": balance
	}, null, completion)
}

// 充值明细(分页)
function queryOfflinePayDetail(userId, page, limit, completion) {
	networking.functionRequest("/queryOfflinePayDetail", null, {
		"userId": userId,
		"current": page,
		"size": limit
	}, null, completion)
}




// 提交usdt提现申请
function submitUSDTWithdraw(userId, coin_number, arrival_amount, usdt_fee, out_address, completion) {
	networking.functionRequest("/submitUSDTWithdraw", null, {
		"userId": userId,
		"coin_number": coin_number,
		"arrival_amount": arrival_amount,
		"usdt_fee": usdt_fee,
		"out_address": out_address
	}, null, completion)
}

// 提现申请
function withdrawCash(userId, cashNumber, paymentCode, completion) {
	networking.functionRequest("/withdrawCash", null, {
		"userId": userId,
		"cashNumber": cashNumber,
		"paymentCode": paymentCode
	}, null, completion)
}

// 金币划转
function giftCoinTransfer(userId, balance, completion) {
	networking.functionRequest("/giftCoinTransfer", null, {
		"userId": userId,
		"balance": balance
	}, null, completion)
}

// 超级存储——金币划转
function vipgiftCoinTransfer(userId, balance, completion) {
	networking.functionRequest("/vipgiftCoinTransfer", null, {
		"userId": userId,
		"balance": balance
	}, null, completion)
}

module.exports = {
	queryBalance: queryBalance,
	transfer: transfer,
	offlineRecharge: offlineRecharge,
	queryOfflinePayDetail: queryOfflinePayDetail,
	submitUSDTWithdraw: submitUSDTWithdraw,
	withdrawCash: withdrawCash,
	giftCoinTransfer: giftCoinTransfer,
	vipgiftCoinTransfer: vipgiftCoinTransfer
}
